### PR TITLE
Update JPA ORM Diagnostics Trace to utilize function for Introspector…

### PR DESCRIPTION
**Pull request by jgrassel**
Signed-off-by: Joe Grassel <jgrassel@us.ibm.com>

Updated the JPA ORM Diagnostic collector on the trace side of things, utilizing the code updates which were earlier dropped in for the dump introspector.

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java [hunk 1]
```diff
@@ -26,30 +26,116 @@
...
 import com.ibm.ws.jpa.diagnostics.JPAORMDiagnostics;
 import com.ibm.ws.jpa.diagnostics.class_scanner.ano.jaxb.classinfo10.ClassInfoType;
-import com.ibm.ws.jpa.diagnostics.puscanner.PersistenceUnitScannerResults2;
+import com.ibm.ws.jpa.diagnostics.puscanner.PersistenceUnitScannerResults;
 
 /**
  *
  */
 public class JPAIntrospection {
+    private static final TraceComponent tc = Tr.register(JPAORMDiagnostics.class,
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java]: This is the wrong class. Should be JPAIntrospection.class

File: dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java [hunk 2]
```diff
@@ -166,92 +252,133 @@ public void executeIntrospectionAnalysis(final PrintWriter dout) {
...
+                sb.append(baos.toString());
+
+                Tr.debug(tc, "JPAORMDiagnostics Dump", sb.toString());
+            }
+        } catch (Throwable t) {
+            FFDCFilter.processException(t, JPAIntrospection.class.getName() + ".executeTraceAnalysis", "185");
+        }
+    }
+
+    private void doExecuteIntrospectionAnalysis(final PrintWriter dout) {
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java]: Should methods like this have a try/catch to ensure any error we encounter during introspection does not prevent the application itself from working?  Seems that an application that could start and run just fine may fail to start if we have a bug in some of our introspection code, like this method.

File: dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java (hunk 1 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java]: Ah!  Great catch, thanks!

File: dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java (hunk 2 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java]: Good point, wrapping the static public method body impls with a catch block that logs to ffdc.

**Comment by jgrassel** [comment]
Pushed a new commit with review suggestions.

**Comment by jgrassel** [comment]
Test failure looks unrelated.  Merging.

**Pull request merged**